src/EnergyPlus/WaterUse.cc

```diff
@@ -133,24 +127,23 @@ namespace WaterUse {
         if (!state.dataGlobal->BeginEnvrnFlag) state.dataWaterUse->MyEnvrnFlagLocal = true;
 
         // Simulate all unconnected WATER USE EQUIPMENT objects
-        for (WaterEquipNum = 1; WaterEquipNum <= state.dataWaterUse->numWaterEquipment; ++WaterEquipNum) {
-            if (state.dataWaterUse->WaterEquipment(WaterEquipNum).Connections == 0) {
-                state.dataWaterUse->WaterEquipment(WaterEquipNum).CalcEquipmentFlowRates(state);
-                state.dataWaterUse->WaterEquipment(WaterEquipNum).CalcEquipmentDrainTemp(state);
+        for (auto &waterEquipment : state.dataWaterUse->WaterEquipment) {
```

Comment:
Yes!  For loops where we do not care about the index at all, and simply want to iterate over every object in a list, using range based loops with references to the objects helps avoid accidental indexing problems, and lets us rely on compilers for handling the underlying looping.  I know they may not always be preferred, but in this use case, they are awesome and simple.